runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -517,8 +518,15 @@ public synchronized void invalidateLayout() {
                         // it is acceptable (at least the code on 10/13/2016 does not have issues)
                         // but setEpoch of routers needs to be synchronized as those variables are
                         // not local.
-                        l.getAllServers().stream().map(getRouterFunction).forEach(x ->
-                                x.setEpoch(l.getEpoch()));
+                        try {
+                            l.getAllServers().stream().map(getRouterFunction).forEach(x ->
+                                    x.setEpoch(l.getEpoch()));
+                        } catch (NetworkException ne) {
```

Comment:
Are you expecting getRouterFunction to throw a NetworkException?

Reply:
Yes, the NettyClientRouter can throw this on creation if its unable to create a connection.